### PR TITLE
refactor(core): add users to mls conv

### PR DESCRIPTION
**Pull request by PatrykBuniX**
Some improvements to adding users to MLS group related code.

### CODE REVIEW

File: packages/core/src/conversation/ConversationService/ConversationService.ts [hunk 1]
```diff
@@ -286,13 +286,20 @@ export class ConversationService {
...
+   */
   public async addUsersToMLSConversation({
-    qualifiedUserIds,
+    qualifiedUsers,
     groupId,
     conversationId,
   }: Required<AddUsersParams>): Promise<MLSReturnType> {
     const groupIdBytes = Decoder.fromBase64(groupId).asBytes;
-    const coreCryptoKeyPackagesPayload = await this.mlsService.getKeyPackagesPayload([...qualifiedUserIds]);
+    const coreCryptoKeyPackagesPayload = await this.mlsService.getKeyPackagesPayload(qualifiedUsers);
```
Comment (PatrykBuniX) [inline packages/core/src/conversation/ConversationService/ConversationService.ts:302]: Unnecessary spread

File: packages/core/src/conversation/ConversationService/ConversationService.types.ts [hunk 2]
```diff
@@ -100,11 +100,11 @@ export type SendMlsMessageParams = SendCommonParams & {
   protocol: ConversationProtocol.MLS;
 };
 
-export type QualifiedUsers = QualifiedId & {skipOwn?: string};
+export type KeyPackageClaimUser = QualifiedId & {skipOwnClientId?: string};
```
Comment (PatrykBuniX) [inline packages/core/src/conversation/ConversationService/ConversationService.types.ts:103]: QualifiedUser seems to be too generic name, with new name we know what this type is being used for.
`skipOwn` -> `skipOwnClientId` - it was hard to say what type of value we expect here, it did sound more like a boolean value.

File: packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts [hunk 3]
```diff
@@ -172,17 +172,17 @@ export class MLSService extends TypedEventEmitter<Events> {
...
      * @note We need to fetch key packages for all the users
      * we want to add to the new MLS conversations,
      * includes self user too.
      */
-    const keyPackages = await Promise.all([
-      ...qualifiedUsers.map(({id, domain, skipOwn}) =>
-        this.apiClient.api.client.claimMLSKeyPackages(id, domain, skipOwn),
+    const keyPackages = await Promise.all(
+      qualifiedUsers.map(({id, domain, skipOwnClientId}) =>
+        this.apiClient.api.client.claimMLSKeyPackages(id, domain, skipOwnClientId),
```
Comment (PatrykBuniX) [inline packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts:183]: map already returns new array, no need to "copy" it again with spread

**Review by atomrc — APPROVED** [review]
seems solid 👏

**Pull request merged**